src/main/java/com/oltpbenchmark/Phase.java

```diff
@@ -187,7 +182,7 @@ public int chooseTransaction(boolean isColdQuery) {
             for (int i = 0; i < this.weightCount; i++) {
                 weight += weights.get(i);
                 if (randomPercentage <= weight) {
-                    return i + 1;
+                    return i;
```

Comment:
Is this a bugfix?

Reply:
i've broken something here that needs to be fixed before this PR can be accepted.  I need some help to better understand phases.  I think i've broken the link between `weight` and `transaction` by introducing transactionid collision.  its seems like we could simplify things here but i'll need some guidance

Reply:
I probably won't be able to take a more detailed pass on this until the weekend